src/citrine/resources/project.py

```diff
@@ -365,3 +365,117 @@ def register(self, name: str, description: Optional[str] = None) -> Project:
 
         """
         return super().register(Project(name, description))
+
+
+    def search(self, search_params: Optional[dict] = None, 
+                page: Optional[int] = None, per_page: int = 100) -> Iterable[Project]:
+        """
+        Search for projects matching the desired name, description, and/or status by exact
+        match or substring match, as specified by the search_params argument. Defaults to
+        no search criteria.
+        
+        Like list, this method allows for pagination. This differs from the list function, 
+        because it makes a POST request to resourceType/search with search fields in a post body.
+
+        Leaving page and per_page as default values will yield all elements in the
+        collection, paginating over all available pages.
+
+        Leaving search_params as its default value will return mimic the behavior of 
+        a full list with no search parameters.
+
+        Parameters
+        ---------
+        search_params: dict, optional
+            A dict representing the body of the post request that will be sent to the
+            search endpoint to filter the results ie.
+            {
+                "search_params": {
+                    "name": {
+                        "value": "Polymer",
+                        "search_method": "SUBSTRING"
+                    },
+                    "description": {
+                        "value": "polymer chain length",
+                        "search_method": "SUBSTRING"
+                    },
+                    "status": {
+                        "value": "not-started",
+                        "search_method": "EXACT"
+                    }
+                }
+            }
+            The dict can constain any combination of (one or all) search specifications
+            for the name, description, and status fields of a project. For each parameter
+            specified, the "value" to match, as well as the "search_method" must be provided.
+            The available search_methods are "SUBSTRING" and "EXACT". The example above demonstrates
+            the input necessary to list projects with names including the word "Polymer,"
+            descriptions containing the phrase "polymer chain length," and statuses exactly
+            equal to "not-started."
+        page: int, optional
+            The "page" of results to list. Default is to read all pages and yield
+            all results.  This option is deprecated.
+        per_page: int, optional
```

Comment:
Is this also deprecated?  If so, we can just leave it out.

Reply:
Oh good point! I'll leave it out!

Reply:
@maxhutch I'm having some issues removing the "page" parameter, because the pagination mechanism in the Pagination class assumes that there is a "page" parameter. Can we leave it in?

Reply:
Can't you just pass in `page = None`?

Reply:
The issue is that then `[this line](https://github.com/CitrineInformatics/citrine-python/blob/1b75914484b4a9138ef7ec7e5b18eced57da8375/src/citrine/_rest/paginator.py#L57)` gets passed "page", but page_fetcher does not have a "page" argument for search if we remove the argument. I tried passing page_kwarg = { 'page': page } if page is Not None else {} like this:  page_fetcher(**page_kwarg, per_page=per_page), but then tests that were expecting to get a "page" argument all fail